src/ppom-controller.ts

```diff
@@ -702,39 +699,51 @@ export class PPOMController extends BaseControllerV2<
     return response;
   }
 
+  /*
+   * Function sends a HEAD request to version info file and compares the ETag to the one saved in controller state.
+   * If ETag is not changed we can be sure that there is not change in files and we do not need to fetch data again.
+   */
+  async #versionInfoETagChanged(url: string): Promise<boolean> {
```

Comment:
This method name can be improved.. maybe `checkIfVersionInfoETagChanged`? Although it also updates the controller state if the etag has changed. So maybe something else?